### PR TITLE
feat(artcentral): Build adapter

### PR DESCRIPTION
**Pull request by 1marcghannam**

### CODE REVIEW

File: packages/sources/artcentral/test/integration/tami.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { assertError, assertSuccess } from '@chainlink/ea-test-helpers'
```
Comment (vnavascues) [inline packages/sources/artcentral/test/integration/tami.ts:1]: Please, always run `yarn lint`:
image

File: packages/sources/artcentral/schemas/env.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "$id": "https://external-adapters.chainlinklabs.com/schemas/artcentral-adapter.json",
+  "title": "ArtCentral External Adapter",
+  "description": "The external adapter for [ArtCentral](https://artcentral.io/)",
+  "required": ["API_KEY"],
+  "type": "object",
+  "properties": {
+    "API_KEY": {
+      "type": "string",
+      "description": "API key taht can be obtained from [ArtCentral](https://artcentral.io/)",
```
Comment (vnavascues) [inline packages/sources/artcentral/schemas/env.json]: `that`

File: packages/sources/artcentral/src/endpoint/tami.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import type { ResponseSchemaTami as ResponseSchema } from '../lib/types'
+import { WEI_MULTIPLIER } from '../lib/const'
+
+export const supportedEndpoints = ['tami']
+
+export const endpointResultPaths = {}
+
+const customError = (data: ResponseSchema) => {
+  const isError = data.TAMI == undefined
```
Comment (vnavascues) [inline packages/sources/artcentral/src/endpoint/tami.ts:13]: Idk whether done in purpose or not, but with `==` also covers the `null` case, which is OK

File: packages/sources/artcentral/src/endpoint/tami.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  const options = { ...config.api, url }
+
+  const response = await Requester.request<ResponseSchema>(options, customError)
+
+  const result = response.data.TAMI * WEI_MULTIPLIER
+
+  if (isNaN(result)) {
+    throw new AdapterError({
+      statusCode: 200,
+      message: `There was an unexpected error parsing the result.`,
```
Comment (vnavascues) [inline packages/sources/artcentral/src/endpoint/tami.ts]: Clearly include the `response.data.TAMI` in the message.

File: packages/sources/artcentral/src/lib/const.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const WEI_MULTIPLIER = 1000000000000000000
```
Comment (vnavascues) [inline packages/sources/artcentral/src/lib/const.ts]: VS Code should warn you about:
```
Numeric literals with absolute values equal to 2^53 or greater are too large to be represented accurately as integers.ts(80008)
```

More on `tami.ts` comments

File: packages/sources/artcentral/src/endpoint/tami.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  const jobRunID = validator.validated.id
+  const assetAddress = validator.validated.data.assetAddress
+
+  const url = util.buildUrlPath(`/collectibles/:assetAddress/appraisal`, { assetAddress })
+
+  const options = { ...config.api, url }
+
+  const response = await Requester.request<ResponseSchema>(options, customError)
+
+  const result = response.data.TAMI * WEI_MULTIPLIER
```
Comment (vnavascues) [inline packages/sources/artcentral/src/endpoint/tami.ts]: BUG: Please, remember than in JavaScript the `MAX_SAFE_INTEGER` is `9007199254740991` (`2^53 - 1`)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

`WEI_MULTIPLIER > MAX_SAFE_INTEGER`, which means `const result = response.data.TAMI * WEI_MULTIPLIER` is not safe.

Install ethers.js and use `BigNumber` to format `response.data.TAMI` in `wei` (you won't need the `WEI_MULTIPLIER`).
https://docs.ethers.io/v5/api/utils/display-logic/#utils-parseUnits

Then return it as result either as `number` or `string`. I think there is no problem encoding a JSON `string` to `uint256` at TOML level (unless Golang can't handle exponents, e.g. `e+23`)

File: packages/sources/artcentral/src/endpoint/tami.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  },
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+
+  const jobRunID = validator.validated.id
+  const assetAddress = validator.validated.data.assetAddress
+
+  const url = util.buildUrlPath(`/collectibles/:assetAddress/appraisal`, { assetAddress })
```
Comment (vnavascues) [inline packages/sources/artcentral/src/endpoint/tami.ts:41]: BUG: `assetAddress` requires to be lower cased. The API is case sensitive and does not accept any other format than lower case addresses (at least tested for Azuki). Sending the addresses as bytes and then decoding them via `base64-decoder` will always return a lower case address, but the adapter alone should make sure addresses are lower cased.

**Review by vnavascues — CHANGES_REQUESTED** [review]
Please, pay attention to big numbers and addresses format, we've talked about them many times. Always have in mind the `MAX_SAFE_INTEGER`. Always test the API with different address formats, e.g. lowercase, checksum, mixed case.

File: packages/sources/artcentral/src/endpoint/tami.ts (hunk 6 above)
Comment (1marcghannam) [reply packages/sources/artcentral/src/endpoint/tami.ts]: I used ethers to format to wei and returned it as a number. I tried returning as string, encoding in the job spec, `"152107753147022770000000"` encoded returned `0x9a91eb0d000000000000000000000000000000000000000000002035c71f41cd39a55080`. My task looked like this `encode [type="ethabiencode"
        abi="fulfillRequest(uint256 result)"
        data="{\\"result\\": $(result_parse)}"
        ]`
       The correct encoding to uint256 would be the same without `9a91eb0d` at the start.

**Pull request merged**